app/Http/Controllers/API/v1/TransportController.php

```diff
@@ -370,25 +371,40 @@ public function create(Request $request): JsonResponse {
                                             'destination' => ['required', 'numeric'],
                                             'departure'   => ['required', 'date'],
                                             'arrival'     => ['required', 'date'],
-                                            'force'       => ['nullable', 'boolean']
+                                            'force'       => ['nullable', 'boolean'],
+                                            'with'        => ['nullable', 'array', 'max:10'],
                                         ]);
+        if (isset($validated['with'])) {
+            $withUsers = User::whereIn('id', $validated['with'])->get();
+            foreach ($withUsers as $user) {
+                if (!Auth::user()?->can('checkin', $user)) {
+                    return $this->sendError('You are not allowed to checkin for the given user.', 403);
+                }
+            }
+        }
 
         try {
             $searchKey          = empty($validated['ibnr']) ? 'id' : 'ibnr';
+            $trip               = HafasController::getHafasTrip($validated['tripId'], $validated['lineName']);
             $originStation      = Station::where($searchKey, $validated['start'])->first();
+            $departure          = Carbon::parse($validated['departure']);
             $destinationStation = Station::where($searchKey, $validated['destination'])->first();
+            $arrival            = Carbon::parse($validated['arrival']);
+            $travelReason       = Business::tryFrom($validated['business'] ?? Business::PRIVATE->value);
+            $event              = isset($validated['eventId']) ? Event::find($validated['eventId']) : null;
 
+            // check in the authenticated user
```

Comment:
Needs to be refactored with the new hydrator

Reply:
Kannst du das bitte machen?